app/services/supplier_locations/importer.rb

```diff
@@ -0,0 +1,57 @@
+# frozen_string_literal: true
+
+module SupplierLocations
+  class Importer
+    DATE_FORMAT = '%Y-%m-%d'
+
+    attr_accessor :filename
+
+    def initialize(filename)
+      self.filename = filename
+    end
+
+    def call
+      message("Updating supplier locations from #{filename}")
+      validate_file!
+
+      message("Updating locations for #{suppliers.size} suppliers, effective from #{effective_from || 'the dawn of time'} to #{effective_to || 'forever and ever'}")
+
+      suppliers.each do |supplier_name, codes|
+        supplier = Supplier.find_by!(key: supplier_name)
+        locations = codes.collect { |code| Location.find_by(nomis_agency_id: code) }.compact
+
+        message("Updating #{locations.size} locations for supplier #{supplier.name}...")
+        SupplierLocation.link_locations(effective_from: effective_from, effective_to: effective_to, supplier: supplier, locations: locations)
+      end
+
+      message('Done.')
+    end
+
+  private
+
+    def yaml
+      @yaml ||= ActiveSupport::HashWithIndifferentAccess.new(YAML.safe_load(File.read(filename)))
+    end
+
+    def suppliers
+      @suppliers ||= yaml[:suppliers]
+    end
+
+    def effective_from
+      @effective_from ||= yaml[:effective_from].presence && Date.strptime(yaml[:effective_from], DATE_FORMAT)
```

Comment:
I don't think we need to get fiddly with dates, here? Can't we just do `Date.parse`?